diag/attribute_error_diagnostic.go

```diff
@@ -0,0 +1,41 @@
+package diag
```

Comment:
@kmoe not sure if you have any opinions here -- I lean slightly towards keeping this self-contained functionality separate to reduce the cognitive burden of looking at the documentation for one large Go package, but happy to combine if consensus pulls there.

Reply:
(Oh sorry I didn't explicitly answer this -- I think import cycles would be a definite possibility since these types get referenced in `tfsdk` and `internal/reflect`.)

Reply:
Thinking on it, I've come around to the conclusion that you're probably right. Let's leave it here for now, and we'll move it later if we want to/can.